### PR TITLE
feat(tabs): exposes new values in brn-tabs-trigger

### PR DESCRIPTION
**Pull request by thatsamsonkid**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] switch
- [ ] table
- [x] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Brn Trigger currently does not expose the underlying elementRef as well as some other key values that could be useful in customizing the tabs component.

See this as an example of code used to create below tabs example

https://github.com/thatsamsonkid/portfolio-analog/blob/main/src/app/content/experience.component.ts#L199

Closes #

## What is the new behavior?

PR update Brn trigger directive to expose underlying elementRef as well getters for key and if the tab is currently active.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

https://github.com/goetzrobin/spartan/assets/22568206/c8bdaa91-9af5-49fe-acea-c98510c329ef

### CODE REVIEW

File: libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { BrnTabsDirective } from './brn-tabs.directive';
 })
 export class BrnTabsTriggerDirective {
 	private _root = inject(BrnTabsDirective);
-	private _elementRef = inject(ElementRef);
+	public elementRef = inject(ElementRef);
```
Comment (goetzrobin) [inline libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:22]: can we also make these readonly?

File: libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts [hunk 2]
```diff
@@ -48,4 +48,12 @@ export class BrnTabsTriggerDirective {
 		if (!this._key) return;
 		this._root.setValue(this._key);
 	}
+
+	get isSelected(): boolean {
```
Comment (goetzrobin) [inline libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:52]: maybe instead change it to a signal called _selected and expose a public readonly selected that is _selected.asReadonly()?

**Review by goetzrobin — CHANGES_REQUESTED** [review]
Left a couple comments. Curious to hear your thoughts

**Comment by goetzrobin** [comment]
Also, @thatsamsonkid  can you add the exposedAs for this issue: https://github.com/goetzrobin/spartan/issues/105? Thanks already!

**Comment by goetzrobin** [comment]
Also, @thatsamsonkid  can you add the exposedAs for this issue: https://github.com/goetzrobin/spartan/issues/105? Thanks already!

**Comment by thatsamsonkid** [comment]
All sounds like good ideas to me, I'll make the changes later today

File: libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts (hunk 1 above)
Comment (thatsamsonkid) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts]: Pushed Change

File: libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts (hunk 2 above)
Comment (thatsamsonkid) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:53]: @goetzrobin - I was just seeing computed signals are readonly by default. So could we not just do the following changes. See this PR https://github.com/thatsamsonkid/spartan/pull/2/commits/8b4676861ced76fc43e98e2931dcf09c3be7b539
Let me know what you think

Comment (goetzrobin) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:53]: I like it! Much cleaner! Tabs was one of the earlier components so I am glad we are revisiting this and cleaning things up!

Comment (thatsamsonkid) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts]: Awesome merged those changes in

File: libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts [hunk 3]
```diff
@@ -8,24 +8,26 @@ import { BrnTabsDirective } from './brn-tabs.directive';
...
-		'[tabindex]': '_isSelected() ? "0": "-1"',
-		'[attr.aria-selected]': '_isSelected()',
+		'[tabindex]': 'selected() ? "0": "-1"',
+		'[attr.aria-selected]': 'selected()',
 		'[attr.aria-controls]': 'contentId',
-		'[attr.data-state]': "_isSelected() ? 'active' : 'inactive'",
+		'[attr.data-state]': "selected() ? 'active' : 'inactive'",
 		'[attr.data-orientation]': '_orientation()',
 		'[attr.data-disabled]': "disabled ? '' : undefined",
 		'(click)': 'activate()',
```
Comment (goetzrobin) [inline libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: We should probably start adding the `exportAs` for all of the brains 🧠 Thoughts?

Comment (goetzrobin) [inline libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: We should probably start adding the `exportAs` for all of the brains 🧠 Thoughts?

Comment (goetzrobin) [inline libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: We should probably start adding the `exportAs` for all of the brains 🧠 Thoughts?

Comment (thatsamsonkid) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: Yea we definitely can. I think makes sense for some use cases where you just need some simple values to use in the template. Is probably just good to do to make them ever more flexible

Comment (goetzrobin) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: sweet!

Comment (goetzrobin) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: can you add a Closes #105 in the commit for this? Thanks already!

Comment (thatsamsonkid) [reply libs/ui/tabs/brain/src/lib/brn-tabs-trigger.directive.ts:17]: Done and added export as to rest of brn tab directives. Let me know if anything else

**Review by goetzrobin — APPROVED** [review]
LGTM!!

**Pull request merged**